### PR TITLE
Truncating Membership ID when greater than 63 character

### PR DESCRIPTION
**Pull request by akshaybathija-github**
Modifying the Cluster Membership ID to be truncated in case of >63 character based on API limits 

API Limit Reference : https://cloud.google.com/anthos/fleet-management/docs/reference/rest/v1/projects.locations.memberships

Earlier the Formula for Calculation was : ${var.project_id}-${var.location}-${var.cluster_name}
Now in case this exceed 63 character it will trim the ending to fit 63 characters

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@akshaybathija-github
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
modules/fleet-membership/main.tf
--- old/modules/fleet-membership/main.tf
+++ new/modules/fleet-membership/main.tf
@@ -15,9 +15,9 @@
  */
 
 locals {
-  hub_project_id          = var.hub_project_id == "" ? var.project_id : var.hub_project_id
+  hub_project_id                   = var.hub_project_id == "" ? var.project_id : var.hub_project_id
   gke_hub_membership_name_complete = var.membership_name != "" ? var.membership_name : "${var.project_id}-${var.location}-${var.cluster_name}"
-  gke_hub_membership_name = length(local.gke_hub_membership_name_complete)>63 ? substr(local.gke_hub_membership_name_complete,0,63) : local.gke_hub_membership_name_complete
+  gke_hub_membership_name          = length(local.gke_hub_membership_name_complete) > 63 ? substr(local.gke_hub_membership_name_complete, 0, 63) : local.gke_hub_membership_name_complete
 }
 
 # Retrieve GKE cluster info
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

**Comment by comment-bot-dev** [comment]
@akshaybathija-github
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@akshaybathija-github
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: modules/fleet-membership/main.tf [hunk 1]
```diff
@@ -15,8 +15,9 @@
  */
 
 locals {
-  hub_project_id          = var.hub_project_id == "" ? var.project_id : var.hub_project_id
-  gke_hub_membership_name = var.membership_name != "" ? var.membership_name : "${var.project_id}-${var.location}-${var.cluster_name}"
+  hub_project_id                   = var.hub_project_id == "" ? var.project_id : var.hub_project_id
+  gke_hub_membership_name_complete = var.membership_name != "" ? var.membership_name : "${var.project_id}-${var.location}-${var.cluster_name}"
+  gke_hub_membership_name          = length(local.gke_hub_membership_name_complete) > 63 ? substr(local.gke_hub_membership_name_complete, 0, 63) : local.gke_hub_membership_name_complete
```
Comment (bharathkkb) [inline modules/fleet-membership/main.tf:20]: nit: we may not need the len check since  substr(string,0,n>len(string) == string anyway.

**Review by bharathkkb — COMMENTED** [review]
Thanks for the PR @akshaybathija-github

File: modules/fleet-membership/main.tf (hunk 1 above)
Comment (bharathkkb) [reply modules/fleet-membership/main.tf:20]: Also looks like it per API docs it must end with an alphanumeric character. Let's proactively wrap a trimsuffix(substr..,"-") if for some reason the result of substr turns out to end in "-".

Comment (akshaybathija-github) [reply modules/fleet-membership/main.tf]: fixed

**Comment by comment-bot-dev** [comment]
@akshaybathija-github
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@akshaybathija-github
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Pull request merged**